src/main/java/org/cbioportal/cdd/service/internal/CDDServiceTopBraidImpl.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2018 - 2020 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+package org.cbioportal.cdd.service.internal;
+
+import java.util.*;
+import org.cbioportal.cdd.model.CancerStudy;
+import org.cbioportal.cdd.model.ClinicalAttributeMetadata;
+import org.cbioportal.cdd.service.ClinicalDataDictionaryService;
+import org.cbioportal.cdd.service.exception.CancerStudyNotFoundException;
+import org.cbioportal.cdd.service.exception.ClinicalAttributeNotFoundException;
+import org.cbioportal.cdd.service.exception.ClinicalMetadataSourceUnresponsiveException;
+import org.cbioportal.cdd.service.exception.FailedCacheRefreshException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+/**
+ * Note this class relies on the ClinicalAttributeMetadataCache class which will frequently generate a
+ * new cache of clinical attributes.  Each method in this class should only get the cache once
+ * and should not attempt to make any modifications to the cache.
+ *
+ * @author Robert Sheridan, Avery Wang, Manda Wilson
+ */
+@Service
+@Qualifier("topbraid")
+public class CDDServiceTopBraidImpl implements ClinicalDataDictionaryService {
+
+    @Autowired
+    private ClinicalAttributeMetadataCache clinicalAttributesCache;
+
+    @Autowired
+    private LevenshteinDistanceCache levenshteinDistanceCache;
```

Comment:
This is deprecated now ... but can be updated in a future PR (work has been done already)

Reply:
we can make a card for future work